TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -2,169 +2,127 @@ const Parser = require( './Parser' );
-            } else if (!BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]) {
-                isValid = false;
-            // Benchmark should have at least one metric to parse with
-            } else if (!BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]
-            || !BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]["metrics"]
-            || Object.keys(BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]]["metrics"]).length === 0) {
-                isValid = false;
-            }
-    
-            if ( ( curRegexResult = productResourceRegex.exec( curItr.value ) ) !== null ) {
-                curProductResource = curRegexResult[1];
+    static parsePerf(results) {
+        if(Array.isArray(results) && results.length > 0) {
+            for(let result of results) {
+                if (this.canParse(result.testName)) {
+                    const {benchmarkName, benchmarkVariant, testData} = this.parse(result.testName, result.testOutput, result.testResult);
+                    result.benchmarkName = benchmarkName;
+                    result.benchmarkVariant = benchmarkVariant;
+                    result.testData = testData;
+                }
             }
+        }
+        return results;
+    }
 
-            if ( isValid ) {
-
-                curBenchVariant = BenchmarkMetric[BenchmarkMetricRouter[curBenchmarkName][curBenchmarkVariant]];	
-                curSearchString = curItr.value;
-                
-                // if outerRegex is undefined, all runs should be measured. Parse metrics in every run
-                // if outerRegex is defined, any runs before outerRegex will be ignored. Parse metrics in warm runs only 
-                if (curBenchVariant.outerRegex !== undefined) {
-                	if ( ( curRegexResult = curBenchVariant.outerRegex.exec( curSearchString ) ) !== null ) {
-                        // index 0 contains entire text (curItr.value)
-                        // index 1 contains text after the outerRegex
-                        curSearchString = curRegexResult[1];
-                    } 
+    static parse( testName, testOutput, testResult ) {
+        const {benchmarkName, benchmarkVariant} = this.getBenchmarkInfo(testName);
+        let curTestData = {};
+        let curBenchVariant = null;
+        let curMetric = null;
+        let curRegexResult = null;
+        let curMetricValues = null;
+        let curRegex = null;
+
+        // if (!BenchmarkMetricRouter[benchmarkName] || !BenchmarkMetricRouter[benchmarkName][benchmarkVariant] || !BenchmarkMetric[BenchmarkMetricRouter[benchmarkName][benchmarkVariant]]
+        // || !BenchmarkMetric[BenchmarkMetricRouter[benchmarkName][benchmarkVariant]]["metrics"]
+        // || Object.keys(BenchmarkMetric[BenchmarkMetricRouter[benchmarkName][benchmarkVariant]]["metrics"]).length === 0) {
+        //     isValid = false;
+        // }
```

Comment:
We still need some of the checks there. 
We don't need `!BenchmarkMetricRouter[benchmarkName] || !BenchmarkMetricRouter[benchmarkName][benchmarkVariant]` since it's done in `canParse()`, but we do need rest of the checks.